src/handlers/defaultPropsHandler.js

```diff
@@ -118,7 +124,10 @@ export default function defaultPropsHandler(
 ) {
   let statelessProps = null;
   const defaultPropsPath = getDefaultPropsPath(componentDefinition);
-  if (isStatelessComponent(componentDefinition)) {
+  /**
+   * function, lazy, memo, forwardRef etc components can resolve default props as well
+   */
+  if (!isReactComponentClass(componentDefinition)) {
     statelessProps = getStatelessPropsPath(componentDefinition);
```

Comment:
Happy to change that too. This was always a bit misleading and with hooks it is definitely wrong.

Reply:
What are you referring here? Changing the name?

Reply:
Naming yes. Doesn't matter really since it's internals only. This should probably be handled in a different PR and applied consistently. But I don't know if this would change some react-docgen/utils API so I guess this is fine.

Reply:
What would you rename it to?

Reply:
I don't know. Probably just getFunctionPropsPath or just getPropsPath. Naming is hard but I don't think this discussion serves any purpose.